chain/xrp/builder/builder.go

```diff
@@ -0,0 +1,93 @@
+package builder
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	xc "github.com/cordialsys/crosschain"
+	"github.com/cordialsys/crosschain/chain/xrp/tx"
+	xrptxinput "github.com/cordialsys/crosschain/chain/xrp/tx_input"
+	binarycodec "github.com/xyield/xrpl-go/binary-codec"
+)
+
+// TxBuilder for Template
+type TxBuilder struct {
+	Asset xc.ITask
+}
+
+var _ xc.TxBuilder = &TxBuilder{}
+
+type TxInput = xrptxinput.TxInput
+
+// NewTxBuilder creates a new Template TxBuilder
+func NewTxBuilder(cfgI xc.ITask) (xc.TxBuilder, error) {
+	return TxBuilder{
+		Asset: cfgI,
+	}, nil
+}
+
+// NewTransfer creates a new transfer for an Asset, either native or token
+func (txBuilder TxBuilder) NewTransfer(from xc.Address, to xc.Address, amount xc.AmountBlockchain, input xc.TxInput) (xc.Tx, error) {
+	if _, ok := txBuilder.Asset.(*xc.TokenAssetConfig); ok {
+		return txBuilder.NewTokenTransfer(from, to, amount, input)
+	}
+	return txBuilder.NewNativeTransfer(from, to, amount, input)
+}
+
+// NewNativeTransfer creates a new transfer for a native asset
+func (txBuilder TxBuilder) NewNativeTransfer(from xc.Address, to xc.Address, amount xc.AmountBlockchain, input xc.TxInput) (xc.Tx, error) {
+	txInput := input.(*TxInput)
+
+	signingPubKeyHex := hex.EncodeToString(txInput.Pubkey)
+	//txInput.XRPTx.SigningPubKey = signingPubKeyHex
+	fmt.Println(txInput)
+
+	jsonData, err := json.Marshal(txInput.XRPTx)
+	if err != nil {
+		return nil, errors.New("error marshalling struct")
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(jsonData, &result)
+	if err != nil {
+		return nil, errors.New("error unmarshalling struct")
+	}
+
+	if sequence, ok := result["Sequence"].(float64); ok {
+		result["Sequence"] = int(sequence)
+	}
+
+	if flags, ok := result["Flags"].(float64); ok {
+		result["Flags"] = int(flags)
+	}
+
+	if lastLedgerSequence, ok := result["LastLedgerSequence"].(float64); ok {
+		result["LastLedgerSequence"] = int(lastLedgerSequence)
+	}
```

Comment:
if we defined our inputs explicitly (integer `Sequence` and `LastLedgerSequence`, then we wouldn't have to do this.

Reply:
I don't think it will work as:

"
 // To unmarshal JSON into an interface value,
// Unmarshal stores one of these in the interface value:
//
//   - bool, for JSON booleans
//   - float64, for JSON numbers
//   - string, for JSON strings
//   - []interface{}, for JSON arrays
//   - map[string]interface{}, for JSON objects
//   - nil for JSON null
//
"
.
Values are defined corect but the unmarshal operation changes them.

Reply:
There's no need to go through json unmarshaling.
1. Explicitly define the inputs required in the TxInput.

```go
type TxInput struct {
  // ...
   Sequence int
   LastLedgerSequence int
  // ...
}
```

2. Assign values to transaction object in TxBuilder.

```
result["Sequence"] = txInput.Sequence
result["LastLedgerSequence"] = txInput.LastLedgerSequence
```

No unmarshaling or conversion to floats needed.